src/msw/utils.cpp

```diff
@@ -1166,23 +1166,37 @@ wxLoadUserResource(const wxString& resourceName,
 namespace
 {
 
-// Helper function wrapping Windows GetVersionEx() which is deprecated since
-// Windows 8. For now, all we do in this wrapper is to avoid the deprecation
-// warnings but this is not enough as the function now actually doesn't return
-// the correct value any more and we need to use VerifyVersionInfo() to perform
-// binary search to find the real Windows version.
-OSVERSIONINFOEX wxGetWindowsVersionInfo()
+// Helper trying to get the real Windows version which is needed because
+// GetVersionEx() doesn't return it any more since Windows 8.
+OSVERSIONINFOEXW wxGetWindowsVersionInfo()
 {
-    OSVERSIONINFOEX info;
+    OSVERSIONINFOEXW info;
     wxZeroMemory(info);
+    info.dwOSVersionInfoSize = sizeof(info);
+
+    // The simplest way to get the version is to call the kernel
+    // RtlGetVersion() directly, if it is available.
+#if wxUSE_DYNLIB_CLASS
+    wxDynamicLibrary dllNtDll;
+    if ( dllNtDll.Load(wxS("ntdll.dll"), wxDL_VERBATIM | wxDL_QUIET) )
+    {
+        typedef LONG /* NTSTATUS */ (WINAPI *RtlGetVersion_t)(OSVERSIONINFOEXW*);
+
+        RtlGetVersion_t wxDL_INIT_FUNC(pfn, RtlGetVersion, dllNtDll);
+        if ( pfnRtlGetVersion &&
+                (pfnRtlGetVersion(&info) == 0 /* STATUS_SUCCESS */) )
+        {
+            return info;
+        }
+    }
+#endif // wxUSE_DYNLIB_CLASS
 
 #ifdef __VISUALC__
     #pragma warning(push)
     #pragma warning(disable:4996) // 'xxx': was declared deprecated
 #endif
 
-    info.dwOSVersionInfoSize = sizeof(info);
-    if ( !::GetVersionEx(reinterpret_cast<OSVERSIONINFO *>(&info)) )
+    if ( !::GetVersionExW(reinterpret_cast<OSVERSIONINFOW *>(&info)) )
```

Comment:
We still support Win9x in 3.0 branch and this wouldn't work there, would it? Or does MSLU take care of it (I didn't use it since 15 years or so, so I don't remember at all how does it work any more...)?

Reply:
Didn't really think about Win9x, and that is to long ago to remember if MSLU would handle that.

Reply:
I don't know what to do about this, chances of breaking something look very small, but OTOH this is not the kind of change we're supposed to do in the stable branch.

Maybe we could load `GetVersionExW()` dynamically? This certainly would be safe... Or just the `A` version for this one?

Reply:
To support Win9x the `A` version is probably necessary as to my understanding the `W` version is not provided by the OS. But it would complicate the code a bit because the struct could not be shared between the kernel method and `GetVersionEx`.